src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const setOfStates = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        setOfStates.push(currentState);
```

Comment:
The assignment to `currentState` and pushing it to `setOfStates` is being repeated in each case. Consider moving this logic outside the switch statement to adhere to the DRY principle.